modules/vectra.py

```diff
@@ -2475,3 +2494,753 @@ def delete_account_note(self, account_id=None, note_id=None):
+        Set to empty list to clear tags (default: False)
+        """
+        return super().set_account_tags(account_id, tags, append)
+
+    @renew_access_token
+    def bulk_set_accounts_tag(self, tag, account_ids):
+        """
+        Set a tag in bulk on multiple accounts. Only one tag can be set at a time
+        :param account_ids: IDs of the accounts for which to set the tag
+        """
+        return super().bulk_set_accounts_tag(tag, account_ids)
+
+    @renew_access_token
+    def bulk_delete_accounts_tag(self, tag, account_ids):
+        """
+        Delete a tag in bulk on multiple accounts. Only one tag can be deleted at a time
+        :param account_ids: IDs of the accounts on which to delete the tag
+        """
+        return super().bulk_delete_accounts_tag(tag, account_ids)
+
+    @renew_access_token
+    def get_account_note(self, account_id=None):
+        """
+        Get account notes
+        :param account_id: account ID
+        """
+        return super().get_account_note(account_id)
+
+    @renew_access_token
+    def set_account_note(self, account_id=None, note=''):
+        """
+        Set account note
+        :param account_id: account ID
+        :param note: content of the note to set
+        """
+        return super().set_account_note(account_id, note)
+
+    @renew_access_token
+    def update_account_note(self, account_id=None, note_id=None, note=''):
+        """
+        Set account note
+        :param account_id: account ID
+        :param note_id: ID of the note to update
+        :param note: updated content of the note
+        """
+        return super().update_account_note(account_id, note_id, note)
+    
+    @renew_access_token
+    def delete_account_note(self, account_id=None, note_id=None):
+        """
+        Set account note
+        :param account_id: account ID
+        :param note_id: ID of the note to delete
+        """
+
+        return super().delete_account_note(account_id, note_id)
+
+    def get_all_rules(self, **kwargs):
+        """
+        Generator to retrieve all rules page by page - all parameters are optional
+        :param contains: search for rules containing this string (substring matching)
+        :param fields: comma separated string of fields to be filtered and returned
+            possible values are: active_detections, additional_conditions, created_timestamp,
+            description, detection, detection_category, enabled, id, is_whitelist, last_timestamp,
+            priority, source_conditions, template, total_detections, triage_category, url
+        :param include_templates: include rule templates, default is False
+        :param ordering: field used to sort response
+        :param page: page number to return (int)
+        :param page_size: number of object to return in repsonse (int)
+        """
+        resp = self._get_request(url='{url}/rules'.format(url=self.url), **self._generate_rule_params(kwargs))
```

Comment:
We should be able to leverage the parent classes methods here.  We can add a _get_request method to the base client class (or v2).  Then you could tailor the _get_request to the specific API version.  That would make the code easier to maintain.